### PR TITLE
Group list

### LINKED ISSUE #84: "Group you manage" page: the group list
As you can see on `http://dev.algorea.org/branch/design/`, when clicking on "groups you managed" without selecting any group, there is a page which summarizes everything on group you manage. 

- implement the service fetching the groups the user manages (https://france-ioi.github.io/algorea-devdoc/api/#operation/managedGroupsView)
- list the groups in a grid with as columns: name, type, can_manage, can_watch_members,can_grant_group_access
- you can get inspired by the group you joined (`/groups/joined`) and manager list (`/groups/by-id/:id/details/managers`) pages
- while loading, display a loader on the grid
- if the user does not manage any group, display a message "You do not manage any group".
- only display the "direct children", do not include the "direct children / all descendants" on top.
- clicking on a group name navigates to this group page

### PR DESCRIPTION
**Pull request by AyeRlS**
Fixes #84. 

Please note that I have removed the translation due to the problems I have with the xlf generation I have and it's why it is failing to build.

### CODE REVIEW

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  getType(value: GroupType): string {
+    if (value === 'Class') return $localize`Class`;
+    if (value === 'Base') return $localize`Base`;
+    if (value === 'Club') return $localize`Club`;
+    if (value === 'Friends') return $localize`Friends`;
+    if (value === 'Session') return $localize`Session`;
+    if (value === 'Team') return $localize`Team`;
+    return $localize`Other`;
```
Comment (smadbe) [inline src/app/modules/group/components/managed-group-list/managed-group-list.component.ts:51]: if if if... why not a switch case?

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  getCanManage(value: ManageType): string {
+    if (value === 'memberships') return $localize`Memberships`;
+    if (value === 'memberships_and_group') return $localize`Memberships & Group`;
+    return $localize`None`;
```
Comment (smadbe) [inline src/app/modules/group/components/managed-group-list/managed-group-list.component.ts:57]: same

File: src/locale/messages.fr.xlf [hunk 3]
```diff
@@ -35,65 +35,47 @@
           <context context-type="linenumber">14,15</context>
         </context-group>
       </trans-unit><trans-unit id="5402924068010392760" datatype="html">
-        <source>Join the group "<x id="PH" equiv-text="response.group.name"/>"</source><target state="translated">Rejoindre le groupe "<x id="PH" equiv-text="response.group.name"/>"</target>
+        <source>Join the group "<x id="PH" equiv-text="        header: $lo"/>"</source><target state="translated">Rejoindre le groupe "<x id="PH" equiv-text="response.group.name"/>"</target>
```
Comment (smadbe) [inline src/locale/messages.fr.xlf:38]: this change seems strange and unrelated
same for all this file...

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts (hunk 1 above)
Comment (AyeRlS) [reply src/app/modules/group/components/managed-group-list/managed-group-list.component.ts]: I find switch not really readable and tend to not use them if I don't need the fall-through behavior, but I've changed the code to use a switch

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts (hunk 2 above)
Comment (AyeRlS) [reply src/app/modules/group/components/managed-group-list/managed-group-list.component.ts]: Fixed.

File: src/locale/messages.fr.xlf (hunk 3 above)
Comment (AyeRlS) [reply src/locale/messages.fr.xlf:38]: running `npm run extract-i18n` does that on my computer so I don't really know what to do

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        this.state = state.tag;
+        if (isReady(state)) this.data = state.data;
+      },
+      _err => this.state = 'error',
+    );
+  }
+
+  ngOnInit(): void {
+    this.dataFetching.next();
+  }
```
[the remark this answers is not in the document]
Comment (AyeRlS) [reply src/app/modules/group/components/managed-group-list/managed-group-list.component.ts]: Fixed.

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts [hunk 5]
```diff
@@ -16,29 +14,23 @@ export class ManagedGroupListComponent implements OnDestroy, OnInit {
...
-        this.state = state.tag;
-        if (isReady(state)) this.data = state.data;
-      },
-      _err => this.state = 'error',
-    );
   }
 
   ngOnInit(): void {
-    this.dataFetching.next();
+    this.subscription = this.managedGroupService.getManagedGroups().pipe().subscribe(
```
Comment (smadbe) [inline src/app/modules/group/components/managed-group-list/managed-group-list.component.ts:23]: There is no subscription to keep on an http service, it cannot last forever.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/locale/messages.fr.xlf (hunk 3 above)
Comment (smadbe) [reply src/locale/messages.fr.xlf:38]: Without changing anything by hand, just running `npm run extract-i18n`? If you have this `        header: $lo` for id="5402924068010392760" in the `messages.xlf`,  it is strange. Do you have version 11 of `ng` installed (`ng --version`)?

Anyway, you should never push something which is not related in a PR.

File: src/app/modules/group/components/managed-group-list/managed-group-list.component.ts (hunk 5 above)
Comment (smadbe) [reply src/app/modules/group/components/managed-group-list/managed-group-list.component.ts:23]: Probably you mixed with placed where there is a subscription to be able to cancel it when a refresh occurs.

Comment (AyeRlS) [reply src/app/modules/group/components/managed-group-list/managed-group-list.component.ts]: Fixed.

File: src/locale/messages.fr.xlf (hunk 3 above)
Comment (AyeRlS) [reply src/locale/messages.fr.xlf]: I've removed this whole file and I will let you do the translation.

**Review by smadbe — APPROVED** [review]

**Pull request merged**